### PR TITLE
Remove `cancel_pending_market`

### PR DESCRIPTION
**Pull request by maltekliemann**
Fixes zeitgeistpm/runtime-audit-1#5.

This removes the ability of a normal user to cancel a proposal for a market. This is similar to how proposals for the Polkadot Treasury cannot be withdrawn.

### CODE REVIEW

**Review by sea212 — COMMENTED** [review]
Please add the information that this function and corresponding event was removed in [docs/changelog_for_devs.md](https://github.com/zeitgeistpm/zeitgeist/blob/main/docs/changelog_for_devs.md)

We should also remove those functions from the weight file, otherwise there are just dangling around unused:
https://github.com/zeitgeistpm/zeitgeist/blob/9014b203f41dd7a0ecb2a19ba6ebb86a3663bc60/zrml/prediction-markets/src/weights.rs#L36
https://github.com/zeitgeistpm/zeitgeist/blob/9014b203f41dd7a0ecb2a19ba6ebb86a3663bc60/zrml/prediction-markets/src/weights.rs#L116-L120

**Review by Chralt98 — COMMENTED** [review]
The functionality remove of `cancel_pending_market` should have no missing dependencies. People are now hyper-incentivised to create an interesting (approvable) market, otherwise their `AdvisoryBond` gets slashed by the Advisory Committee. 

Idea: But why not allow the root origin to slash (reject) and give the ApprovalOrigin the ability to cancel multiple pending markets (return the AdvisoryDeposits to the market creators)? I mean at some point the blockchain should be incentivised to clear multiple pending markets (storage), if they are low quality and with a big pending queue. To give the power of slashing to the Advisory Committee is a little bit much, don't you think?

File: zrml/prediction-markets/src/weights.rs [hunk 1]
```diff
@@ -33,7 +33,6 @@ pub trait WeightInfoZeitgeist {
     fn admin_move_market_to_resolved_overhead() -> Weight;
     fn approve_market() -> Weight;
     fn buy_complete_set(a: u32) -> Weight;
-    fn cancel_pending_market() -> Weight;
     fn create_categorical_market() -> Weight;
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply zrml/prediction-markets/src/weights.rs:36]: Yeah, `weights.rs` is auto-generated. No point in editing this, I think.

**Comment by maltekliemann** [comment]
Thanks for the review! I see your point about the risk that users taking a risk when proposing a market.

> People are now hyper-incentivised to create an interesting (approvable) market, otherwise their AdvisoryBond gets slashed by the Advisory Committee. [...] To give the power of slashing to the Advisory Committee is a little bit much, don't you think?

I don't think so. I wouldn't call this _hyper-incentivised_. We need to make sure that markets on Zeitgeist are sound. Otherwise we maneuver ourselves into difficult situations.

The other thing is that we need to make sure that the advisory queue is not spammed. Of course, proposing a market costs transaction fees, which prevent that the chain gets DoSed. But the queue is handled by a small amount of humans, so even small amounts of spam are a problem. Hence, we need to make it a punishable offence to propose bad markets. To put it differently: Slashing the advisory bond is the only power that the advisory comittee has.

**Pull request merged**